internal/server/server_test.go

```diff
@@ -98,3 +98,48 @@ func TestSOMASServerFactoryInitialisesClients(t *testing.T) {
 		}
 	}
 }
+
+func TestRandomUniqueAssignIIGORoles(t *testing.T) {
```

Comment:
as this depends on rand I'd seed the rng:
```
	rand.Seed(time.Now().UnixNano())
```

Reply:
I wrote the tests so they'd work with the random. I read that it was difficult to tests functions with random numbers as the tests are threaded? 
Is what I've done okay, or should I try to update it?

Reply:
> it was difficult to tests functions with random numbers as the tests are threaded?

Yes it is indeed hard to test with rand stuff. Not really because of the threaded reason (because we currently don't run them in parallel)

Usually we would inject the `rand` functions we use in the function (e.g. if we use `rand.Shuffle`, we'd pass in instead as a parameter a `shuffler` function that we can mock in the tests.

This is a simple test for relatively simple functionality--seeding the RNG by time is fine--it might cause flaky tests, but eh, at this point, I'm fine with that.